Core/src/org/sleuthkit/autopsy/casemodule/services/TagNameDefiniton.java

```diff
@@ -156,40 +177,55 @@ private String toSettingsFormat() {
+        List<String> standardTags = new ArrayList<>(STANDARD_TAG_DISPLAY_NAMES);  //modifiable copy of default tags list for us to keep track of which ones already exist
         String setting = ModuleSettings.getConfigSetting(TAGS_SETTINGS_NAME, TAG_NAMES_SETTING_KEY);
         if (null != setting && !setting.isEmpty()) {
-            List<String> notableTags = null;
             List<String> tagNameTuples = Arrays.asList(setting.split(";"));
-            for (String tagNameTuple : tagNameTuples) {
-                String[] tagNameAttributes = tagNameTuple.split(",");
-                if (tagNameAttributes.length == 3) { //Upgrade case Tags.properties does not contain any tag definitions with knownStatus
-                    if (notableTags == null) {
-                        String badTagsStr = ModuleSettings.getConfigSetting("CentralRepository", "db.badTags"); // NON-NLS
-                        if (badTagsStr == null) { //if a badtags list could not be read from a central repository properties file use the defualt bad tags
-                            badTagsStr = STANDARD_NOTABLE_TAG_DISPLAY_NAMES;
-                        }
-                        if (badTagsStr.isEmpty()) { //if a badtags list is empty the user has saved all tags as non-notable so we will have all tags non-notable
-                            notableTags = new ArrayList<>();
-                        } else {
-                            notableTags = new ArrayList<>(Arrays.asList(badTagsStr.split(","))); //if the badtags list was present and had contents use the contents as the current notable tags list
-                        }
-                    }
-                    if (notableTags.contains(tagNameAttributes[0])) { //if the name attribute is in the notable tags list add the tag as a notable tag
+            List<String> notableTags = new ArrayList<>();
+            String badTagsStr = ModuleSettings.getConfigSetting("CentralRepository", "db.badTags"); // NON-NLS
+            if (badTagsStr == null || badTagsStr.isEmpty()) {  //if there were no bad tags in the central repo properties file use the default list
+                notableTags.addAll(STANDARD_NOTABLE_TAG_DISPLAY_NAMES);
+            } else {  //otherwise use the list that was in the central repository properties file
+                notableTags.addAll(Arrays.asList(badTagsStr.split(",")));
+            }
+            for (String tagNameTuple : tagNameTuples) { //for each tag listed in the tags properties file
+                String[] tagNameAttributes = tagNameTuple.split(","); //get the attributes
+                if (tagNameAttributes.length == 3) {  //if there are only 3 attributes so Tags.properties does not contain any tag definitions with knownStatus
+                    standardTags.remove(tagNameAttributes[0]);  //remove tag from default tags we need to create still
+                    if (notableTags.contains(tagNameAttributes[0])) { //if tag should be notable mark create it as such
```

Comment:
Perhaps we can split this code into two methods? I know there could be duplication, but I'm thinking the logic would be easier to follow if we checked the tuple length and then branched into either a method to do the upgrade from the Central Repo properties, or just parse the tag properties file. It would also mean that we would not need to read the Central Repo properties file more than once.